lang/en/docs/cli/check.md

```diff
@@ -6,8 +6,14 @@ layout: guide
 
 ##### `yarn check` <a class="toc" id="toc-yarn-check" href="#toc-yarn-check"></a>
 
-Verifies that versions of the package dependencies in the current project's `package.json` matches that of yarn's lock file.
+Verifies that versions of the package dependencies in the current project's `package.json` match those in yarn's lock file.
+
+The switches `--integrity` and `--verify-tree` are mutually exclusive.
 
 ##### `yarn check --integrity` <a class="toc" id="toc-yarn-check-integrity" href="#toc-yarn-check-integrity"></a>
 
-Verifies that versions and hashed value of the package contents in the project's `package.json` matches that of yarn's lock file. This helps to verify that the package dependencies have not been altered.
+Verifies that versions and hashed values of the package contents in the project's `package.json` match those in yarn's lock file. This helps to verify that the package dependencies have not been altered.
+
+##### `yarn check --verify-tree` <a class="toc" id="toc-yarn-check-verify-tree" href="#toc-yarn-check-verify-tree"></a>
+
+Recursively verifies that the dependencies in `package.json` are present in `node_modules` and have the right version. This check does not consider `yarn.lock`.
```

Comment:
This is where the heavy lifting happens: https://github.com/yarnpkg/yarn/blob/9d678b9f996dbc3234295f8b0220f0f04fa23421/src/cli/commands/check.js#L85-L143